src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/DistinctiveItem.java

```diff
@@ -0,0 +1,31 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+import org.bukkit.inventory.meta.ItemMeta;
+
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Implement this interface for any {@link SlimefunItem} to prevent
+ * cargo using only its ID when comparing. #canStack is used when
+ * comparing stacks
+ *
+ * @author Sefiraat
+ */
+public interface DistinctiveItem extends ItemAttribute {
+
+    /**
+     * This method is called by {@link SlimefunUtils#isItemSimilar} when two {@link SlimefunItemStack}
+     * IDs match on a DistinctiveItem and should return if the two items can stack
+     * with one another.
+     *
+     * @param itemMetaOne
+     *                    The {@link ItemMeta} of the first stack being compared.
+     * @param itemMetaTwo
+     *                    The {@link ItemMeta} of the second stack being compared.
+     *
+     * @return Whether the two {@link ItemMeta}s are distinct
+     */
+    boolean canStack(@Nonnull ItemMeta itemMetaOne, @Nonnull ItemMeta itemMetaTwo);
```

Comment:
I'm wondering if we want SlimefunItem or at least ItemStack here rather than Meta. Feels like that expands the realm of possibility a lot more. We can also change later but that of course means a deprecation cycle.

Reply:
IIRC my thought process was to provide the itemmeta to stop people pulling out the itemmeta’s again should they need to adding more overhead to an already poorly performant process. I could provide the stacks alongside however and make it clear in the docs? I could also form up a compound object to wrap that all into or something?

Reply:
Yeah good thought not wanting to add overhead. We know how... fun getItemMeta is...

Yeah I think 2 params is best.
Not sure a wrapper is needed, I think that's probably just more annoying tbh.

Reply:
For the purposes here the items only really ‘merge’ and dupe when the slimefunitems are the same and just something within the meta makes them distinct be it lore *yuck* or PDC being the common factor. So hopefully this covers most use cases, it certainly does for all the known ones atm.

Reply:
In which case I’ll implement the feedback tomorrow with the aim to merge asap!